src/code/reducers/helpers/load-state-from-authoring.js

```diff
@@ -76,28 +98,45 @@ function processAuthoredDrakes(authoredChallenge, authoredTrialNumber, template,
     }
     // Keep the drake as female until the end, so no sex-linked information is lost for linked drakes
     let femaleDrake = new BioLogica.Organism(BioLogica.Species.Drake, drakeDef.alleles, BioLogica.FEMALE);
+    const actualDrakeSex = drakeDef.sex != null
+                            ? drakeDef.sex
+                            : Math.round(Math.random());
 
+    /*
+     * Synchronize the linkedGenes
+     */
     alleleString = femaleDrake.getAlleleString();
-    if (authoredChallenge.linkedGenes) {
-      let linkedGenesDef = authoredChallenge.linkedGenes;
-      if (Array.isArray(authoredChallenge.linkedGenes)) {
-        linkedGenesDef = authoredChallenge.linkedGenes[authoredTrialNumber];
+    const { linkedGenes } = authoredChallenge;
+    if (linkedGenes) {
+      let linkedGenesDef = linkedGenes;
+      if (Array.isArray(linkedGenes)) {
+        linkedGenesDef = linkedGenes[authoredTrialNumber];
       }
       if (i === linkedGenesDef.drakes[0]) {
-        linkedGeneDrake = femaleDrake;
+        linkedGeneDrake = new BioLogica.Organism(BioLogica.Species.Drake, femaleDrake.getAlleleString(), actualDrakeSex);
       } else if (linkedGenesDef.drakes.indexOf(i)) {
-        let linkedGenes = split(linkedGenesDef.genes);
-        for (let gene of linkedGenes) {
+        let genes = split(linkedGenesDef.genes);
+        for (let gene of genes) {
           let copyIntoGenes = femaleDrake.genetics.genotype.getAlleleString([gene], femaleDrake.genetics);
           let masterGenes = linkedGeneDrake.genetics.genotype.getAlleleString([gene], femaleDrake.genetics);
           alleleString = alleleString.replace(copyIntoGenes, masterGenes);
         }
       }
+    
+      /*
+       * Guarantee minimum number of moves
+       */
+      const { linkedGenes: { minChanges }, userChangeableGenes } = authoredChallenge;
+      if ((i > 0) && minChanges && userChangeableGenes) {
+        const userGenes = split(userChangeableGenes);
+        while (numberOfMoves(linkedGeneDrake, alleleString, actualDrakeSex) < minChanges) {
```

Comment:
Are we safe from an infinite loop situation? Make sure minChanges is validated to acceptable range just in case of authoring glitch.

Reply:
Good thought! I added the additional constraints to the schema.